### PR TITLE
[SofaPython] Regroup dependencies

### PR DESCRIPTION
**Pull request by fredroy**
SofaPython was needed into SofaTest (to enable python test) and SofaExporter for bindings.

1- then CMake had to enable SofaPython then SofaTest then SofaPython/SofaPython_test, which is quite "bothersome",and made the main SOFA Cmakefile cluttered.
Now if SofaTest is enabled, SofaPython will enable the use of Python_test (and not the other way around)
(NB SofaPython contains a redirection for project including `SofaTest/Python_test.h`)

2- SofaExporter is a Sofa module and so, it is quite weird to have a dependency to a plugin in a module.
Now SofaPython will enable SofaExporter bindings if SofaExporter has been enabled before  (and not the other way around)

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

**Comment by fredroy** [comment]
[ci-build][with-all-tests][force-full-build]

File: applications/plugins/SofaPython/CMakeLists.txt [hunk 1]
```diff
@@ -141,13 +141,43 @@ set(PYTHON_FILES
...
 )
 
-
 find_package(PythonLibs 2.7 REQUIRED)
 find_package(SofaGui REQUIRED)
 find_package(SofaUserInteraction REQUIRED)
 find_package(SofaMiscMapping REQUIRED)
 find_package(Eigen3 REQUIRED)
 
+sofa_find_package(SofaTest QUIET)
```
Comment (damienmarchal) [inline applications/plugins/SofaPython/CMakeLists.txt:150]: find_package or sofa_find_package :)

**Comment by damienmarchal** [comment]
This sounds nice and mandatory...I'm looking how did the inversion of dependencies.

EDIT: ok i saw

File: applications/plugins/SofaPython/CMakeLists.txt (hunk 1 above)
Comment (fredroy) [reply applications/plugins/SofaPython/CMakeLists.txt:150]: sofa_find_package to get the cmake variable SOFAPYTHON_HAVE_SOFATEST defined in the cmake.in, etc

Comment (damienmarchal) [reply applications/plugins/SofaPython/CMakeLists.txt:150]: So all the other find_package in every CMakeLists.txt should become sofa_find_package ?

**Comment by guparan** [comment]
I propagated SofaPython dep changes in more plugins.
I also added a few quick-cleans, see my nice commit messages to get the idea :+1:

**Pull request merged**